src/pages/index.js

```diff
@@ -73,25 +115,124 @@ addCardButton.addEventListener("click", () => {
 function handleProfileFormSubmit(inputValues) {
-  userProfileInformation.setUserInfo(inputValues);
-  profileFormPopup.close();
+  profileFormPopup.setLoading(true);
+  api
+    .updateUserInformation(inputValues)
+    .then((result) => {
+      return result.json();
+    })
+    .then((userInformation) => {
+      userProfileInformation.setUserInfo(userInformation);
+      profileFormPopup.setLoading(false);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(profileFormPopup.close());
 }
 
 function createCard(element) {
-  const card = new Card(element, "#photos-template", handleImageClick);
+  const card = new Card(
+    element,
+    "#photos-template",
+    handleImageClick,
+    handleCardDelete,
+    handleLikeCard
+  );
   return card.generateCard();
 }
 
 function handleImageClick({ name, link }) {
   photoCardPopup.open({ name, link });
 }
 
+// Submitting a new card to the server
 function handleCardFormSubmit(inputValues) {
-  renderCard(inputValues);
-  cardFormPopup.close();
+  cardFormPopup.setLoading(true);
+  api
+    .addCard(inputValues)
+    .then((result) => result.json())
+    .then((newCard) => {
+      renderCard(newCard);
+      cardFormPopup.setLoading(false);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(cardFormPopup.close());
 }
 
 function renderCard(item, method = "prepend") {
   cardSection.addItem(item);
 }
+
+//Delete card
+function handleCardDelete(card) {
+  const cardId = card.getCardId();
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmit(() => {
+    deleteCardPopup.setLoading(true);
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        card.deleteCard();
+        deleteCardPopup.setLoading(false);
+      })
+      .catch((error) => {
+        console.error("Error in deleting card:", error);
+      })
+      .finally(deleteCardPopup.close());
+  });
+}
+
+//handles both like and dislike of cards
+function handleLikeCard(card) {
+  const cardId = card.getCardId();
+  const cardLiked = card.getCardIsLiked();
+  if (cardLiked == false) {
+    api
+      .likeCard(cardId)
+      .then(() => {
+        card.likeCard();
+      })
+      .catch((error) => {
+        console.error("Error in liking card:", error);
+      });
+  } else {
+    api
+      .dislikeCard(cardId)
+      .then(() => {
+        card.dislikeCard();
+      })
+      .catch((error) => {
+        console.error("Error in disliking card: ", error);
+      });
+  }
+}
+
+//updating the avatar
+function handleAvatarUpdate({ name: inputValue }) {
+  api
+    .udpateAvatar(inputValue)
+    .then(() => {
+      userProfileInformation.setUserAvatar(inputValue);
+    })
+    .catch((error) => {
+      console.error("Error in updating avatar: ", error);
+    })
+    .finally(avatarUpdatePopup.close());
+}
+
+// Get user data from the server
+api
+  .getUser()
+  .then((res) => {
+    userProfileInformation.setUserInfo(res);
+  })
+  .catch((err) => {
+    console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors